### DISCUSSION TITLE
Is dud usable on Windows if avoiding symlinks?

### DISCUSSION THREAD

**Discussion by ofekd** [kevin-hanselman/dud#132] [Q&A] [answered]
I would like to adopt dud for a cross-platform app, and can opt to only use copy checkouts on Windows. Are there any other gotchas I should be aware of? Aside for lack of testing, of course.

Thanks!

**Comment by kevin-hanselman** [comment, marked as answer]
Off the top of my head, Windows' lack of symlinks is the primary incompatibility. Feel free to give Dud a shot and see what breaks😄. If you encounter any bugs, you can record them here rather than in Issues, as Windows support remains unplanned.

I'd love to learn more about your use case! Feel free to chat in this thread, or reach out to me or on LinkedIn or Twitter. Thanks for considering Dud!

**Reply by ofekd** [reply]
Thank you for being so responsive.

I am going over the source to see if I can use dud to power a VCS of sorts for filmmaking pipelines. Currently not sure it's possible because I can't find a way to store multiple versions of the same file, as an artifact can only be owned by a single stage.

Can't version using git neither because it's going to be hard to manage the resulting mess of branches. For example: both `branch A` and `branch B` depend on `master` and now I want to see version `commit a` from `branch A` along with version `commit b` from `branch B`. I will now have to create a temp branch, merge them together, do the work, then cherry pick the result to the different branches, which is very cumbersome.

I could store the artifacts' checksums then "trick" dud and replace the checksum in the stage according to the version I want, but that negates some of the advantages of using dud in the first place, not to mention it becoming hard to tell which part of the cache can be safely removed.

I'll update here if I manage to find a solution

**Comment by blakeNaccarato** [comment]
I'm curious about Windows support as well. I'm sure it's a tall order and I understand that it's unplanned currently. Windows does have symbolic links, and in my usage of them, they behave as you would expect. But I'm sure there's some incompatibility between the symlink abstractions on Windows and Linux that I am not personally aware of.

In any case, I'm interested in this tool and will probably play with it in WSL for small projects, but it would be cool to see Windows on the long-term roadmap if this tool really starts taking off!

**Reply by kevin-hanselman** [reply]
I think using Dud in WSL may work fine. And it may work fine in Windows, too. Please feel free to give it a go and report back. If there's only one or two small things that need correcting in Dud to get Windows support working, I'm willing to entertain a "Use at your own risk" level of Windows support.
